config/config/src/auth.ts

```diff
@@ -92,8 +91,20 @@ export function inheritAuthConfig (targetCfg: InheritableConfig, authSrcCfg: Inh
   inheritPickedConfig(targetCfg, authSrcCfg, pickAuthConfig, pickRawAuthConfig)
 }
 
-export const isSupportedNpmConfig = (key: string): boolean =>
-  key.startsWith('@') || key.startsWith('//') || NPM_AUTH_SETTINGS.includes(key)
+const isNpmCompatConfig = (key: string): false | 'compat' =>
+  (PNPM_COMPAT_SETTINGS as string[]).includes(key) && 'compat'
```

Comment:
all this complexity with compat is needed because of a single setting `manage-package-manager-versions`? This setting should work from `pnpm-workspace.yaml`. Don't add special compat logic for it.

Reply:
https://github.com/pnpm/pnpm/pull/10145/commits/9dd4482ebe954e2c5ad6fed898c521ffa88f9e36

Reply:
> I can also remove manage-package-manager-versions entirely. But this would mean that people using older versions of pnpm as a pnpm version manager would not be able to automatically fetch newer pnpm versions as specified in the pnpm-workspace.yaml file.

So this setting is still written to npmrc just for older versions of pnpm to read it? But users can manually add it to npmrc if they really want that. We should not add it to npmrc because npm prints warnings about this settings now.